### PR TITLE
PHP8 Update. Bump nikic/php-parser version and updated functions

### PR DESCRIPTION
**Pull request by ciaran-moore**
Updated php-parser functions PHP8 and updated FunctionParser library.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
   "require": {
     "php": ">=7.4|>=8.0",
     "ext-json": "*",
-    "nikic/php-parser": "^4.4",
-    "jeremeamia/functionparser": "^1.0"
+    "nikic/php-parser": "^5.0",
+    "simpletools/functionparser": "1.0.1"
```
Comment (fieryprophet) [inline composer.json]:
```suggestion
    "simpletools/functionparser": "^1.0.1"
```

Locking it to a specific version is usually bad form in a library, ^ allows future backwards-compatible versions of this library to be used as well.

**Review by fieryprophet — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (ciaran-moore) [reply composer.json]: I have amended the version to `"simpletools/functionparser": "^1.0"` on the pull branch now.

**Comment by ciaran-moore** [comment]
Could a tag be created with the changes from this pull request? I'm installing with `composer require cordeva/php-sandbox` and the latest version is still pulling through the old `functionparser` library.

**Comment by fieryprophet** [comment]
Done, I bumped the version to 3.1 since we are replacing an underlying dependency.

**Comment by ciaran-moore** [comment]
Fantastic, thank you :)

**Pull request merged**